File: src/main/java/com/learn/EmployeeManagementSystem/service/DepartmentService.java
package com.learn.EmployeeManagementSystem.service;

import com.learn.EmployeeManagementSystem.DTOs.requestDto.DepartmentRequest;
import com.learn.EmployeeManagementSystem.DTOs.responseDTO.DepartmentResponse;

import java.util.List;

public interface DepartmentService
{
    // Get all Departments.
    List<DepartmentResponse> getAllDeparments();

    //Get department by Id
    DepartmentResponse getDepartmentById(Long id);

    // Create department
    DepartmentResponse createDepartment(DepartmentRequest departmentRequest);

    // Update department
    DepartmentResponse updateDepartment(Long id, DepartmentRequest departmentRequest);

    // Delete department
    void deletedepartment(Long id);
}

File: frontend/src/services/departmentService.js
import api from './api';

export const departmentService = {
  // GET /departments
  getAllDepartments: async () => {
    const response = await api.get('/departments');
    return response.data;
  },

  // GET /departments/{id}
  getDepartmentById: async (id) => {
    const response = await api.get(`/departments/${id}`);
    return response.data;
  },

  // POST /departments
  createDepartment: async (departmentData) => {
    const response = await api.post('/departments', departmentData);
    return response.data;
  },

  // PUT /departments/{id}
  updateDepartment: async (id, departmentData) => {
    const response = await api.put(`/departments/${id}`, departmentData);
    return response.data;
  },

  // DELETE /departments/{id}
  deleteDepartment: async (id) => {
    const response = await api.delete(`/departments/${id}`);
    return response.data;
  },
};

export default departmentService;
